modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java

```diff
@@ -67,6 +112,228 @@ public void onAfterCreate(Company company) {
+					companyId, modelName, ResourceConstants.SCOPE_GROUP,
+					String.valueOf(userPersonalSiteGroupId), powerUserRoleId,
+					modelActionIds.toArray(new String[0]));
+			}
+		}
+
+		private Role _getPowerUserRole(long companyId) {
+			try {
+				return _roleLocalService.getRole(
+					companyId, RoleConstants.POWER_USER);
+			}
+			catch (PortalException portalException) {
+				_log.error(
+					"Unable to get power user role in company " + companyId,
+					portalException);
+			}
+
+			return null;
+		}
+
+		private Group _getUserPersonalSiteGroup(long companyId) {
+			try {
+				return _groupLocalService.getUserPersonalSiteGroup(companyId);
+			}
+			catch (PortalException portalException) {
+				_log.error(
+					"Unable to get user personal site group in company " +
+						companyId,
+					portalException);
+			}
+
+			return null;
+		}
+
+		private void _initPermissions(Company company, Portlet portlet) {
+			long companyId = company.getCompanyId();
+
+			Role powerUserRole = _getPowerUserRole(companyId);
+
+			if (powerUserRole == null) {
+				return;
+			}
+
+			Group userPersonalSiteGroup = _getUserPersonalSiteGroup(companyId);
+
+			if (userPersonalSiteGroup == null) {
+				return;
+			}
+
+			try {
+				if (companyId == portlet.getCompanyId()) {
+					initPermissions(
+						companyId, powerUserRole.getRoleId(),
+						portlet.getRootPortletId(),
+						userPersonalSiteGroup.getGroupId());
+				}
+			}
+			catch (PortalException portalException) {
+				_log.error(
+					StringBundler.concat(
+						"Unable to initialize user personal site permissions ",
+						"for portlet ", portlet.getPortletId(), " in company ",
+						company.getCompanyId()),
+					portalException);
+			}
+		}
+
+		private class PanelAppServiceTrackerCustomizer
+			implements EagerServiceTrackerCustomizer<PanelApp, PanelApp> {
+
+			public PanelAppServiceTrackerCustomizer(
+				BundleContext bundleContext) {
+
+				_bundleContext = bundleContext;
+			}
+
+			@Override
+			public PanelApp addingService(
+				ServiceReference<PanelApp> serviceReference) {
+
+				PanelApp panelApp = _bundleContext.getService(serviceReference);
+
+				try {
+					Portlet portlet = panelApp.getPortlet();
+
+					if (portlet == null) {
+						portlet = _portletLocalService.getPortletById(
+							panelApp.getPortletId());
+					}
+
+					if (portlet == null) {
+						Class<?> panelAppClass = panelApp.getClass();
+
+						_log.error(
+							StringBundler.concat(
+								"Unable to get portlet ",
+								panelApp.getPortletId(), " for panel app ",
+								panelAppClass.getName()));
+
+						return panelApp;
+					}
+
+					initPermissions(portlet);
+
+					return panelApp;
+				}
+				catch (Throwable throwable) {
+					_bundleContext.ungetService(serviceReference);
+
+					throw throwable;
+				}
+			}
+
+			@Override
+			public void modifiedService(
+				ServiceReference<PanelApp> serviceReference,
+				PanelApp panelApp) {
+
+				removedService(serviceReference, panelApp);
```

Comment:
You don't consume any property from the ServiceReference, there is nothing to modify here.